launch/test/launch/test_execute_process.py

```diff
@@ -168,3 +171,40 @@ def generate_launch_description():
     ls.include_launch_description(generate_launch_description())
     assert 0 == ls.run()
     assert expected_called_count == on_exit_callback.called_count
+
+
+def test_execute_process_prefix_filter_match():
+    lc = LaunchContext()
+    lc._set_asyncio_loop(asyncio.get_event_loop())
+    SetLaunchConfiguration('launch-prefix', 'time').visit(lc)
+    assert len(lc.launch_configurations) == 1
+    SetLaunchConfiguration(
+        'launch-prefix-filter',
+        f'{os.path.basename(sys.executable)}').visit(lc)
+    assert len(lc.launch_configurations) == 2
+
+    test_process = ExecuteProcess(
+        cmd=[sys.executable, '-c', "print('action')"],
+        output='screen'
+    )
+
+    test_process.execute(lc)
+    assert 'time' in test_process.process_details['cmd']
+
+
+def test_execute_process_prefix_filter_no_match():
+    lc = LaunchContext()
+    lc._set_asyncio_loop(asyncio.get_event_loop())
+    SetLaunchConfiguration('launch-prefix', 'time').visit(lc)
+    assert len(lc.launch_configurations) == 1
+    SetLaunchConfiguration(
+        'launch-prefix-filter', 'no-match').visit(lc)
+    assert len(lc.launch_configurations) == 2
+
+    test_process = ExecuteProcess(
+        cmd=[sys.executable, '-c', "print('action')"],
+        output='screen'
+    )
+
+    test_process.execute(lc)
+    assert 'time' not in test_process.process_details['cmd']
```

Comment:
@camm73 it'd be good to have a test where a prefix is specified for `ExecuteProcess` (and make sure it behaves as expected).